main.go

```diff
@@ -89,9 +117,11 @@ func main() {
 		os.Exit(1)
 	}
 
+	go runPrometheusServer(metricsAddr)
```

Comment:
Do we really only want those metrics on leader election?
I think the cache etc. runs anyways, no? So the metrics can be useful :thinking:

Reply:
I mean the controler workers running all the watches etc. I thought they always run as well.
So workqueue length metrics still could matter.

If it's like in kubevirt I think we should also collect all pods. @marceloamaral or @nelsonspbr might have some more thoughts on that.

Reply:
"/metrics" is the end point that Prometheus queries to collect metrics from the pod, and we want to have such an endpoint on every pod, that way we can get the metrics from all of them (and count the number of pods running :))

Reply:
Ok, I'm not familiar with how metrics and prometheus work in openshift. If you think it makes sense for a pod to report metrics even if it looses leader election, then a goroutine is fine.

When an operator pod looses leader election, it will only run webhook service.

If you use a goroutine, can you only wrap the `ListenAndServeTLS()` method in a goroutine, not the whole function `runPrometheusServer()` ?

Reply:
Regarding workqueue metrics for the SSP controller, it might be useful to track. But I am not sure if It could be enabled as we enabled for the internal Kubevirt controllers, as [here](https://github.com/marceloamaral/kubevirt/blob/46a3f061ecb8af5207f7cdc675ea8ec7a05fa79e/pkg/monitoring/workqueue/prometheus/prometheus.go). The workqueue must have a name, and I did not see in the SSP controller a transparent workqueue creation

Reply:
@akrejcir I raised this question with the monitoring team, and it seems our approach in OCP is to gather metrics from all pods. Also - moved the goroutine as you requested. Would like to know why this is better though.

Reply:
In my opinion, this is better for code readability. My reasoning is the following:

When I see a goroutine started like this in a function:
```go
func function() {
    // ... do things ...

    go runPrometheusServer()

    // ... do other things ...
}
```

It looks like a probable bug, because there is no visible way how to control the new goroutine. There is no way to wait for its completion or signal it to stop. I would need to read the implementation of the function `runPrometheusServer()` to know that it is intentional.

Another general reason, that maybe does not apply in this case, is that the caller of this function needs to know that it has to be started as a goroutine. So they need to read the documentation or implementation.


When the goroutine is started inside the function as closure, like in this case:
```go
func runPrometheusServer() {
    server := // ... setup server ...

    go func() {
        err := server.ListenAndServeTLS(...)
        // ... handle error ...
    }()
}
```

It is easier to see what it does, and why it needs to be a goroutine. Also, that it is intentional, that there is no way to control the goroutine.


But this is only my opinion, code readability is subjective.